Utilities/Thread.h

```diff
@@ -225,9 +227,63 @@ class thread_ctrl final
 	}
 
 	// Wait once with timeout. May spuriously return false.
+	template <bool accurate = false>
 	static inline void wait_for(u64 usec, bool alert = true)
 	{
-		_wait_for(usec, alert);
+		if constexpr (!accurate)
+		{
+			_wait_for(usec, alert);
+			return;
+		}
+
+		if (!usec)
+		{
+			return;
+		}
+
+		using namespace std::chrono_literals;
+
+		const auto until = std::chrono::time_point_cast<decltype(1us)>(std::chrono::steady_clock::now()) + decltype(1us)(usec);
```

Comment:
I don't think decltype is necessary here at all. This should be fine:
```C++
const auto until = steady_clock::now() + 1us * usec;
```
Should work